### DISCUSSION TITLE
Proxy traffic leaks when starting a scan

### DISCUSSION THREAD

**Discussion by ocervell** [projectdiscovery/nuclei#3702] [Q&A] [answered]
### Current Behavior:
No matter if I use `-proxy socks5://127.0.0.1:9050` (tor socks5 proxy) or `-proxy http://127.0.0.1:9080` (tor http proxy), I still see some traffic leak in Wireshark when **starting** the Nuclei run.

It is limited to the few seconds at the beginning of the run, after a while there is no more traffic to the target and it correctly goes through the proxy.

It seems to be always to same request pattern so I guess there is some kind of requests that goes to the target before instantiating the proxy ?

I know there is an [issue ](https://github.com/projectdiscovery/nuclei/issues/1439) for some templates but I'm running just one template here that is not in that scope.

### Expected Behavior:
No requests should go directly to the target from the scanning host when using `-proxy`.

### Steps To Reproduce:

1. Run wireshark and start a fresh capture
2. Run a scan with some individual templates:
```
nuclei -u <HOST> -jsonl -proxy socks5://127.0.0.1:9050 -t ~/nuclei-templates/cves/2000/CVE-2000-0114.yaml
```
```
nuclei -u <HOST> -jsonl -proxy http://127.0.0.1:9080  -t ~/nuclei-templates/cves/2023/CVE-2023-0099.yaml
```
3. Look at wireshark output (it shows CLIENT HELLO's at the beginning of each scan.

### Anything else:
Here is a wireshark screenshot:
![2023-05-08 13_12_52-Capturing from Ethernet](https://user-images.githubusercontent.com/9629314/236810185-5344135c-228c-4ce3-af7f-21f6f96c006f.png)

**Comment by ShubhamRasal** [comment]
Yes, @ocervell  you are right,
In the beginning we do make a call to check `nuclei` latest version and [nuclei-templates](https://github.com/projectdiscovery/nuclei-templates)  version and that call is not going through the proxy.
We will try to fix this, but you can disable that call by passing `-duc` flag for now

**Comment by ehsandeep** [comment, marked as answer]
@ocervell [apart of partial coverage of protocol](https://github.com/projectdiscovery/nuclei/issues/1439), in this case it's version check request to https://api.pdtm.sh/api/v1/tools/nuclei which can be disabled using `-duc` option or adding the same into config file.
